cli/consumer_command.go

```diff
@@ -207,6 +209,8 @@ func configureConsumerCommand(app commandHost) {
 	consSub.Flag("ack", "Acknowledge received message").Default("true").BoolVar(&c.ack)
 	consSub.Flag("raw", "Show only the message").Short('r').UnNegatableBoolVar(&c.raw)
 	consSub.Flag("deliver-group", "Deliver group of the consumer").StringVar(&c.deliveryGroup)
+	consSub.Flag("max-wait", "Max time to wait before unsubscribing.").DurationVar(&c.maxSubWait)
+	consSub.Flag("max-count", "Max number of messages to consume before unsubscribing.").IntVar(&c.maxSubCount)
```

Comment:
kind of want to drop `consumer sub`, its a bit gross.  I think `nats sub` with its jetstream features maybe better to focus on

Reply:
let me know if you can add there or maybe sub doesnt do something that c sub does then fine we add it here for now

Reply:
Ah I missed `nats sub` supports JS now. I can move this over there.

Reply:
I see that `nats sub` already supports `count` which is nice. The only issue/diff I see is that if a durable already exists, it does not attempt to bind rather than building up the set of options. I will attempt to cleanly handle that case.

Reply:
^And the issue I see, for example.. I created a push durable with ack explicit, but the options use `AckNone`, otherwise the "upsert" behavior could have worked.

Reply:
sounds good thanks

Reply:
Yeah, neither nats sub nor consumer sub is good enough at this point :)